spectrum/checks.py

```diff
@@ -502,12 +502,12 @@ def with_resource_checking_method(self, method):
     def with_query_string(self, query_string):
         return JournalCheck(self._host, self._resource_checking_method, query_string)
 
-    def article(self, id, has_figures=False, version=None):
+    def article(self, id, has_figures=False, version=None, headers=None):
```

Comment:
Would prefer to set headers in the constructor like for `query_string` so that they don't have to be passed through the various methods, then a separate object `JOURNAL_GOOGLE_SCHOLAR` can be created that configures them differently than `JOURNAL_CDN`.

Reply:
The idea is not just to avoid long parameters list in method delegations but also to favor readability, pushing all details into the configuration like if `JOURNAL_GOOGLE_SCHOLAR.article()` was a Gherkin step